### PR TITLE
Port plot_unit_summary to the new widgets API.

### PR DESCRIPTION
**Pull request by samuelgarcia**

### CODE REVIEW

File: spikeinterface/widgets/matplotlib/unit_summary.py [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+        backend_kwargs = self.update_backend_kwargs(**backend_kwargs)
+        backend_kwargs['num_axes'] = 0
+        backend_kwargs['ax'] = None
+        backend_kwargs['axes'] = None
+        
+        self.make_mpl_figure(**backend_kwargs)
+        
+        # and use custum grid spec
+        fig = self.figure
+        gs = fig.add_gridspec(3, 4)
```
Comment (alejoe91) [inline spikeinterface/widgets/matplotlib/unit_summary.py:34]: In my opinion you should create the gridspec based on `dp.plot_data_unit_locations`, `dp.plot_data_acc`, and `dp.plot_data_amplitudes`

**Comment by samuelgarcia** [comment]
@alejoe91 : ready to review on my side.

File: spikeinterface/exporters/report.py [hunk 2]
```diff
@@ -36,20 +37,46 @@ def export_report(waveform_extractor, output_folder, remove_if_exists=False, for
...
     if we.is_extension('spike_amplitudes'):
-        sac = we.load_extension('spike_amplitudes')
-        amplitudes = sac.get_data(outputs='by_unit')
+        spike_amplitudes = we.load_extension('spike_amplitudes').get_data(outputs='by_unit')
+    elif force_computation:
+        spike_amplitudes = compute_spike_amplitudes(we, peak_sign=peak_sign, outputs='by_unit', **job_kwargs)
     else:
-        amplitudes = compute_spike_amplitudes(we, peak_sign=peak_sign, outputs='by_unit', **job_kwargs)
+        spike_amplitudes = None
+        print('export_report(): no spike_amplitudes exported use compute_spike_amplitudes() if you want then')
```
Comment (alejoe91) [inline spikeinterface/exporters/report.py:56]:
```suggestion
        print('export_report(): spike_amplitudes will not be exported. Use compute_spike_amplitudes() if you want to include them.')
```

File: spikeinterface/exporters/report.py [hunk 3]
```diff
@@ -36,20 +37,46 @@ def export_report(waveform_extractor, output_folder, remove_if_exists=False, for
...
+
+    # load or compute quality_metrics
+    if we.is_extension('quality_metrics'):
+        metrics = we.load_extension('quality_metrics').get_data()
+    elif force_computation:
+        metrics = compute_quality_metrics(we)
+    else:
+        metrics = None
+        print('export_report(): no metrics exported use compute_quality_metrics() if you want then')
```
Comment (alejoe91) [inline spikeinterface/exporters/report.py:65]:
```suggestion
        print('export_report(): quality metrics will not be exported. Use compute_quality_metrics() if you want to include them.')
```

File: spikeinterface/exporters/report.py [hunk 4]
```diff
@@ -36,20 +37,46 @@ def export_report(waveform_extractor, output_folder, remove_if_exists=False, for
...
+
+    # load or compute correlograms
+    if we.is_extension('correlograms'):
+        correlograms, bins = we.load_extension('correlograms').get_data()
+    elif force_computation:
+        correlograms, bins = compute_correlograms(we, window_ms=100.0, bin_ms=1.0)
+    else:
+        correlograms = None
+        print('export_report(): no correlograms exported use compute_correlograms() if you want then')
```
Comment (alejoe91) [inline spikeinterface/exporters/report.py:74]: same here

File: spikeinterface/widgets/matplotlib/all_amplitudes_distributions.py [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+        ax = self.ax
+        
+        unit_amps = []
+        for i, unit_id in enumerate(dp.unit_ids):
+            amps = []
+            for segment_index in range(dp.num_segments):
+                amps.append(dp.amplitudes[segment_index][unit_id])
+            amps = np.concatenate(amps)
+            unit_amps.append(amps)
+        parts = ax.violinplot(unit_amps, showmeans=False, showmedians=False, showextrema=False)
```
Comment (alejoe91) [inline spikeinterface/widgets/matplotlib/all_amplitudes_distributions.py:24]: so it plots all in one figure? even if many units?

File: spikeinterface/widgets/matplotlib/unit_summary.py (hunk 1 above)
Comment (alejoe91) [reply spikeinterface/widgets/matplotlib/unit_summary.py:34]: @samuelgarcia what about this? Otherwise it will leave blank spaces..

Comment (samuelgarcia) [reply spikeinterface/widgets/matplotlib/unit_summary.py:34]: I feel lazy about this but I will try somthing.

Comment (samuelgarcia) [reply spikeinterface/widgets/matplotlib/unit_summary.py]: Done

File: spikeinterface/widgets/matplotlib/all_amplitudes_distributions.py (hunk 5 above)
Comment (samuelgarcia) [reply spikeinterface/widgets/matplotlib/all_amplitudes_distributions.py:24]: Finally I do not think that making multi line stuff will make the figure more visible.
I propose to not export this figure in report when num_units > 100.
No more.

File: spikeinterface/exporters/report.py [hunk 6]
```diff
@@ -92,16 +102,20 @@ def export_report(waveform_extractor, output_folder, remove_if_exists=False, for
...
 
     fig, ax = plt.subplots(figsize=(20, 10))
-    sw.plot_units_depth_vs_amplitude(we, ax=ax, unit_colors=unit_colors)
-    fig.savefig(output_folder / f'units_depth_vs_amplitude.{format}')
+    sw.plot_unit_depths(we, ax=ax, unit_colors=unit_colors)
+    fig.savefig(output_folder / f'unit_depths.{format}')
     if not show_figures:
         plt.close(fig)
+    
+    if spike_amplitudes and len(unit_ids) > 100:
```
Comment (alejoe91) [inline spikeinterface/exporters/report.py:110]: Less than here no?
```suggestion
    if spike_amplitudes and len(unit_ids) < 100:
```

**Review by alejoe91 — CHANGES_REQUESTED** [review]
Last small bug and then good to go for me

**Review by alejoe91 — APPROVED** [review]

**Pull request merged**